File: scrooge-generator/src/main/resources/javagen/readMap.java
TMap _map_{{name}} = _iprot.readMapBegin();
{{name}} = new HashMap<{{keyType}}, {{valueType}}>();
int _i_{{name}} = 0;
{{#keyReadWriteInfo}}
  {{>qualifiedFieldType}} {{keyName}};
{{/keyReadWriteInfo}}
{{#valueReadWriteInfo}}
  {{>qualifiedFieldType}} {{valueName}};
{{/valueReadWriteInfo}}

while (_i_{{name}} < _map_{{name}}.size) {
{{#keyReadWriteInfo}}
  {{>readValue}}
{{/keyReadWriteInfo}}
{{#valueReadWriteInfo}}
  {{>readValue}}
{{/valueReadWriteInfo}}
  {{name}}.put({{keyName}}, {{valueName}});
  _i_{{name}} += 1;
}
_iprot.readMapEnd();

File: scrooge-generator/src/main/resources/scalagen/readMap.scala
val _map = _iprot.readMapBegin()
val _rv = new mutable.HashMap[{{keyType}}, {{valueType}}]
var _i = 0
while (_i < _map.size) {
  val _key = {
{{#keyReadWriteInfo}}
    {{>readValue}}
{{/keyReadWriteInfo}}
  }
  val _value = {
{{#valueReadWriteInfo}}
    {{>readValue}}
{{/valueReadWriteInfo}}
  }
  _rv(_key) = _value
  _i += 1
}
_iprot.readMapEnd()
_rv
